### PR TITLE
Wishlist management

### PR DESCRIPTION
**Pull request by darshandamre**
user can add and remove products from wishlist on products page.
user can see the wishlisted products on wishlist page and remove them from their wishlist.

[Preview](https://52shop-web-git-wishlist-management-darshandamre.vercel.app)

### CODE REVIEW

File: src/pages/Wishlist/Wishlist.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  return (
+    <main className="wishlist mb-8">
+      <h3 className="h3 ta-center mt-3 mb-6">Wishlist</h3>
+
+      {data?.user.wishlist.length === 0 ? (
+        <div className="ta-center mx-2">
+          No items in your wishlist{" "}
+          <Link to="/products" className="link">
+            go to products
+          </Link>{" "}
```
Comment (hrishikeshmane) [inline src/pages/Wishlist/Wishlist.js:19]: remove formatted space

File: src/pages/Wishlist/Wishlist.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+const Wishlist = () => {
+  const { data } = useUserQuery();
+
+  return (
+    <main className="wishlist mb-8">
+      <h3 className="h3 ta-center mt-3 mb-6">Wishlist</h3>
+
+      {data?.user.wishlist.length === 0 ? (
+        <div className="ta-center mx-2">
+          No items in your wishlist{" "}
```
Comment (hrishikeshmane) [inline src/pages/Wishlist/Wishlist.js:16]: remove formatted space

File: src/pages/Wishlist/Wishlist.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+          <Link to="/products" className="link">
+            go to products
+          </Link>{" "}
+          page and add products in your wishlist.
+        </div>
+      ) : null}
```
Comment (hrishikeshmane) [inline src/pages/Wishlist/Wishlist.js:22]: Suggestion: use logical AND (&&) instead of ternary (? :)

**Review by hrishikeshmane — APPROVED** [review]
Rest looks good

File: src/pages/Wishlist/Wishlist.js (hunk 1 above)
Comment (darshandamre) [reply src/pages/Wishlist/Wishlist.js:19]: these are necessary spaces, if i remove them `No items in your wishlistgo to productspage and add....` gets rendered

File: src/pages/Wishlist/Wishlist.js (hunk 2 above)
Comment (darshandamre) [reply src/pages/Wishlist/Wishlist.js:16]: same with this

File: src/pages/Wishlist/Wishlist.js (hunk 3 above)
Comment (darshandamre) [reply src/pages/Wishlist/Wishlist.js:22]: i prefer ternary, sometimes logical AND gives issues so i always use ternary. thanks for your time

File: src/components/Card/Card.js [hunk 4]
```diff
@@ -1,13 +1,34 @@
...
+  useUserQuery
+} from "../../hooks";
 import "./Card.css";
 
 const Card = ({ product }) => {
-  const { name, price, discount, pictureLink, cartItem } = product;
+  const { mutate: addToWishlist } = useAddToWishlistMutation();
+  const { mutate: removeFromWishlist } = useRemoveFromWishlistMutation();
+  const { data } = useUserQuery();
+  const { user } = data ?? {};
```
Comment (rahulyadav139) [inline src/components/Card/Card.js:14]: you can destructure this way: const {data:{user}} = useUserQuery()

**Review by rahulyadav139 — APPROVED** [review]

File: src/components/Card/Card.js (hunk 4 above)
Comment (darshandamre) [reply src/components/Card/Card.js:14]: yes Rahul, but when `data` is `undefined` here, it throws an error, that is why i have destructured like this, thanks

**Pull request merged**